server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -0,0 +1,361 @@
+import org.opensearch.cluster.health.ClusterStateHealth;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.WeightedRouting;
+import org.opensearch.cluster.routing.allocation.AwarenessReplicaBalance;
+import org.opensearch.cluster.routing.allocation.decider.AwarenessAllocationDecider;
+import org.opensearch.common.collect.ImmutableOpenMap;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Cluster state Awareness health information
+ *
+ */
+public class ClusterAwarenessHealth implements Writeable {
+
+    private int numberOfNodes;
+    private int numberOfDataNodes;
+    private int activeShards;
+    private int unassignedShards;
+    private int initializingShards;
+    private int relocatingShards;
+    private int activePrimaryShards;
+    private double activeShardsPercent;
+    private boolean hasDiscoveredClusterManager;
+    private ClusterHealthStatus status;
+    private final ClusterAwarenessAttributeHealth clusterAwarenessAttributeHealth;
+
+    /**
+     * Creates cluster awareness health from cluster state.
+     *
+     * @param clusterState The current cluster state. Must not be null.
+     * @param clusterSettings the current cluster settings.
+     * @param awarenessAttributeName Name of awareness attribute for which we need to see the health
+     */
+    public ClusterAwarenessHealth(ClusterState clusterState, ClusterSettings clusterSettings, String awarenessAttributeName) {
+
+        setClusterHealth(clusterState);
+
+        clusterAwarenessAttributeHealth = getAwarenessAttributeHealth(clusterState, clusterSettings, awarenessAttributeName);
+    }
+
+    private void setClusterHealth(ClusterState clusterState) {
+        ClusterStateHealth clusterStateHealth = new ClusterStateHealth(clusterState);
+        numberOfDataNodes = clusterStateHealth.getNumberOfDataNodes();
+        numberOfNodes = clusterStateHealth.getNumberOfNodes();
+        activeShards = clusterStateHealth.getActiveShards();
+        unassignedShards = clusterStateHealth.getUnassignedShards();
+        initializingShards = clusterStateHealth.getInitializingShards();
+        relocatingShards = clusterStateHealth.getRelocatingShards();
+        status = clusterStateHealth.getStatus();
+        activePrimaryShards = clusterStateHealth.getActivePrimaryShards();
+        activeShardsPercent = clusterStateHealth.getActiveShardsPercent();
+        hasDiscoveredClusterManager = clusterStateHealth.hasDiscoveredClusterManager();
+    }
```

Comment:
Would prefer inline assignment within the ctor and avoid this private method altogether.

Reply:
Do you mean move this within the constructor rather than having a private method? The same was done for better readability standpoint and not to load the cluster.